pkg/cvo/cvo.go

```diff
@@ -627,7 +627,7 @@ func (optr *Operator) upgradeableSync(ctx context.Context, key string) error {
 		return nil
 	}
 
-	return optr.syncUpgradeable()
+	return optr.syncUpgradeable(config)
```

Comment:
The CVO using the in-cluster ClusterVersion to pass state from one goroutine to another feels sticky... I'm not set against it, but it would be nice if we could find a way to plumb the data we need through without having to pass this thing we just got from the cluster here.  `syncUpgradeable` is already an `Operator` method, so it should have access to a lot of state already...

Reply:
I had seen this, but somehow did not spend time figuring out if we can avoid that.

Reply:
Yes, I wasn't keen on it either. However, the `syncAvailableUpdates()` in `pkg/cvo/availableupdates.go` uses the cluster version as a parameter also (https://github.com/openshift/cluster-version-operator/blob/master/pkg/cvo/availableupdates.go#L31), so I ended up using here as well. (Now I see the commit https://github.com/openshift/cluster-version-operator/commit/db150e6db4f6d00c553804f46a531da4ee0f5552 for that change was from 4 years ago).

Another alternative I can think of is calling the 
```go
config, err := optr.cvLister.Get(optr.name)
```
in `syncUpgradeable()` but I'll try to think of a better implementation.

Reply:
I do not think it is a bad idea to pass `config` in `optr.syncUpgradeable(config)` because `upgradeableSync()`  is aleady in a go-routine and it is calling the `optr.cvLister.Get(optr.name)` . So I do not think there is mu h difference between  `upgradeableSync()`  or `SyncUpgradeable()` calling `optr.cvLister.Get(optr.name)`

Reply:
Right, I don't think it matters if we shift the `Get` around.  What I'd like long-term is no get at all in this pathway.  Looking [here][1], [here][2], and [here][3], something like `optr.getUpgradeable().Conditions` will get you the current conditions without having to leave local CVO memory.  Although a lock guarding access to a pointer property seems a bit suspect...

[1]: https://github.com/openshift/cluster-version-operator/blob/f7f9b8d0b34cb6349478869c529ab6cd575a55b2/pkg/cvo/cvo.go#L123-L125
[2]: https://github.com/openshift/cluster-version-operator/blob/f7f9b8d0b34cb6349478869c529ab6cd575a55b2/pkg/cvo/upgradeable.go#L91-L92
[3]: https://github.com/openshift/cluster-version-operator/blob/f7f9b8d0b34cb6349478869c529ab6cd575a55b2/pkg/cvo/upgradeable.go#L143-L149